auth/providers/azure/graph/tokenrefresher.go

```diff
@@ -0,0 +1,22 @@
+/*
+Copyright The Guard Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package graph
+
+// TokenRefresher is an interface to obtain token for MS Graph api
+type TokenRefresher interface {
```

Comment:
The name is a bit confusing that it seems to be using 'refresh_token' in oauth context, but actually it doesn't.

Can consider naming it TokenProvider? And 'Refresh'->'Ensure'?

Reply:
ok. i renamed TokenRefresher -> TokenProvider and Refresh() -> Acquire()